typescript/common-resolvers/src/iog-inference.ts

```diff
@@ -79,52 +130,114 @@ const runIog = async (
   //   ],
   // };
   // End dummy iog
-  if (result) {
-    const geometry = {
-      type: "Polygon",
-      coordinates: result?.polygons,
-    };
-    const { smartToolInput } = await throwIfResolvesToNil(
-      "No label with such id",
-      repository.label.get
-    )({ id: args.data.id }, user);
+  const geometry = {
+    type: "Polygon",
+    coordinates: result?.polygons as number[][][],
+  };
 
-    const xCoordinates = geometry.coordinates.reduce(
-      (xCoordinatesCurrent: number[], polygon: number[][]) => [
-        ...xCoordinatesCurrent,
-        ...polygon.map((point: number[]) => point[0]),
-      ],
-      []
-    );
-    const yCoordinates = geometry.coordinates.reduce(
-      (yCoordinatesCurrent: number[], polygon: number[][]) => [
-        ...yCoordinatesCurrent,
-        ...polygon.map((point: number[]) => point[1]),
-      ],
-      []
-    );
-    const [x, y, X, Y] = [
-      Math.min(...xCoordinates),
-      Math.min(...yCoordinates),
-      Math.max(...xCoordinates),
-      Math.max(...yCoordinates),
-    ];
-    const width = X - x;
-    const height = Y - y;
+  const xCoordinates = geometry.coordinates.reduce(
+    (xCoordinatesCurrent: number[], polygon: number[][]) => [
+      ...xCoordinatesCurrent,
+      ...polygon.map((point: number[]) => point[0]),
+    ],
+    []
+  );
+  const yCoordinates = geometry.coordinates.reduce(
+    (yCoordinatesCurrent: number[], polygon: number[][]) => [
+      ...yCoordinatesCurrent,
+      ...polygon.map((point: number[]) => point[1]),
+    ],
+    []
+  );
+  const [x, y, X, Y] = [
+    Math.min(...xCoordinates),
+    Math.min(...yCoordinates),
+    Math.max(...xCoordinates),
+    Math.max(...yCoordinates),
+  ];
+  const width = X - x;
+  const height = Y - y;
+  return {
+    geometry,
+    x,
+    y,
+    height,
+    width,
+  };
+};
 
-    const now = new Date();
+const createIogLabel = async (
+  _parent: any,
+  args: MutationCreateIogLabelArgs,
+  { repository, user, req }: Context
+) => {
+  const labelId = args?.data?.id ?? uuidv4();
+  // Since we don't have any constraint checks with Dexie
+  // We need to ensure that the imageId and the labelClassId
+  // matches some entity before being able to continue.
+  const image = await throwIfResolvesToNil(
+    `The image id ${args.data.imageId} doesn't exist.`,
+    repository.image.get
+  )({ id: args.data.imageId }, user);
+  const dataUrl = await downloadUrlToDataUrl(image.url, req);
+  const now = new Date();
 
-    const newLabelEntity = {
-      smartToolInput: { ...smartToolInput, ...args.data },
-      updatedAt: now.toISOString(),
-      geometry,
-      x,
-      y,
-      height,
-      width,
-    };
-    await repository.label.update({ id: args.data.id }, newLabelEntity, user);
-  }
+  const { xInit, yInit } = {
+    xInit: Math.min(image.width, Math.max(0, args.data.x)),
+    yInit: Math.min(image.height, Math.max(0, args.data.y)),
+  };
```

Comment:
```suggestion
  const xInit = Math.min(image.width, Math.max(0, args.data.x)),
  const yInit = Math.min(image.height, Math.max(0, args.data.y)),
```

I find it easier to read this way, what do you think?